### PR TITLE
Separatedfiles

### PR DESCRIPTION
**Pull request by PDhanrajnath**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,68 +1,12 @@
...
-};
+import typeDefs from "./schema.js";
+import resolvers from "./resolvers.js";
 
 const server = new ApolloServer({
   typeDefs,
   resolvers,
   context: ({ req }) => {
-    const token = req.headers.authorization;
+    const token = req.headers.authorization || "";
```
Comment (rohithZE) [inline src/index.js]: use [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) instead

File: src/resolvers.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+const headersForAuth = {
+  "Content-Type": "application/json",
+  Authorization: "Bearer " + github_data["token"],
+};
+
+const resolvers = {
+  Query: {
+    getUser: async (_parent, context) => {
+      console.log(context.Authorization);
```
Comment (rohithZE) [inline src/resolvers.js]: remove logs

File: src/resolvers.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      const response = await fetch("https://api.github.com/user", {
```
Comment (rohithZE) [inline src/resolvers.js]: Get the base url from constants/env file
it will be easier to replace in different environments

File: src/resolvers.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    getOrgRepos: async (_parent, args, context) => {
+      const response = await fetch(
+        `https://api.github.com/orgs/${args.orgName}/repos`,
+        {
+          method: "GET",
+          headers: headersForAuth,
+        }
+      );
+      // console.log(await response.json());
```
Comment (rohithZE) [inline src/resolvers.js]: doesn't this method return anything?

Comment (PDhanrajnath) [reply src/resolvers.js]: no Rohith
I'll remove it
in assignment they said to use only one api

File: src/resolvers.js (hunk 3 above)
Comment (PDhanrajnath) [reply src/resolvers.js]: ok Rohith

File: src/index.js [hunk 5]
```diff
@@ -1,69 +1,12 @@
...
+import typeDefs from "./schema.js";
+import resolvers from "./resolvers.js";
 
 const server = new ApolloServer({
   typeDefs,
   resolvers,
   context: ({ req }) => {
-    const token = req.headers.authorization;
-    // console.log(JSON.stringify(token));
+    const token = "" ?? req.headers.authorization;
```
Comment (rohithZE) [inline src/index.js]: it should be reversed.

```suggestion
    const token =  req.headers.authorization ?? "";
```

File: src/resolvers.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import fetch from "node-fetch";
+import { headersForAuth, baseUrl } from "./constants";
+
+const resolvers = {
+  Query: {
+    getUser: async (_parent) => {
+      console.log(process.env.BASE_URL);
```
Comment (rohithZE) [inline src/resolvers.js]: Remove log

File: .gitignore [hunk 7]
```diff
@@ -10,9 +10,10 @@
 
 # production
 /build
-
+/src/constants.js
```
Comment (rohithZE) [inline .gitignore]: you don't have to ignore constants

**Review by rohithZE — APPROVED** [review]

**Pull request merged**